Add `Generics` option to `GolangInfo`

justinvp opened the pull request:
This is a new Go schema option in Pulumi v3.84.0 that should be exposed for bridged providers. Commits:
1. Update deps to v3.84.0
2. Expose the `Generics` option

Fixes #1391

t0yv0 reviewed (approved):
This LGTM. Is there a link to a list of more tasks around generics rollout?

iwahbe commented on pkg/tfbridge/info.go:
```diff
@@ -672,6 +672,13 @@ type GolangInfo struct {
 	// associated module dependencies from the SDK's go.mod.
 	InternalDependencies []string
+
+	// Specifies how to handle generating a variant of the SDK that uses generics.
+	// Allowed values are the following:
+	// - "none": do not generate a generics variant of the SDK
+	// - "side-by-side": generate a side-by-side generics variant of the SDK under the x subdirectory
+	// - "only-generics": generate a generics variant of the SDK only
```
I would love this comment to explain what `""` is interpreted as. 

Otherwise LGTM

justinvp replied:
Updated the comment. `"none"` is currently the default when not specified. Although, at some point in the near future we intend to flip the default to `"side-by-side"` so all providers get a generic SDK by default.

justinvp commented:
Update: I addressed the feedback to add a comment on the default value, but also wanted to upgrade to v3.85.0 as there is a Python sdkgen regression in v3.84.0 that is fixed in v3.85.0. But we've tweaked the `Provider` interface in v3.85.0 and need at react to those changes, which I haven't done yet.

t0yv0 commented:
Y'all don't need to take on updating p/p can delegate to providers; I think https://github.com/pulumi/pulumi-terraform-bridge/actions/runs/6261833982/job/17002725064 the automated upgrade job is not doing it right again, we should fix it up.
